lib/clif-backend/src/code.rs

```diff
@@ -58,8 +58,8 @@ impl ModuleCodeGenerator<CraneliftFunctionCodeGenerator, Caller, CodegenError>
         unimplemented!("cross compilation is not available for clif backend")
     }
 
-    fn backend_id() -> Backend {
-        Backend::Cranelift
+    fn backend_id() -> String {
+        "cranelift".to_string()
```

Comment:
I think it would be better to have typed backend ids. An approach similar to `InternalField` for middlewares might work: https://github.com/wasmerio/wasmer/blob/f2f1a08e07988947042252122048e3155b96bbe6/lib/runtime-core/src/vm.rs#L139

Reply:
We can do this with a wrapper struct and (static) strings too.

I think we'll probably want to have a way to print out the name of the backend in either case, so we'll either need to have the string be part of the `id` or have a mapping from id to string.